services/fetcher/randomAnimeFetcher.js

```diff
@@ -0,0 +1,59 @@
+const axios = require("axios");
+
+const config = {
+  method: "GET",
+  url: "https://api.jikan.moe/v4/random/anime",
+};
+
+const rankLimit = 3000; // will pick animes below the rank of 3000
+
+const fetchRandomAnime = async () => {
+  try {
+    while (true) {
+      const response = await axios.request(config);
+      let anime = {};
+      // anime should be approved by MAL
+      if (response?.data?.data && response.data.data.approved) {
+        const res = response.data.data;
+        if (res.rank > rankLimit) continue; // rank limit check
+        if (res.title) anime.title = res.title;
+        if (res.url) anime.url = res.url;
+        if (res.synopsis) anime.synopsis = res.synopsis;
+        if (res.genres) {
+          const genre = res.genres.map((g) => g.name);
+          anime.genre = genre;
+        }
+        if (res.rank) anime.rank = res.rank;
+        if (typeof res.airing !== "undefined") anime.airing = res.airing;
```

Comment:
why didn't you check this one like others as well?

Reply:
because it is a boolean value. checking this like others will affect the working of the variable.